forms-flow-web/src/components/Form/Item/Edit.js

```diff
@@ -68,191 +61,52 @@ const reducer = (form, { type, value }) => {
-  const [saveAsNewVersionselected, setSaveAsNewVersion] = useState(false);
-  const [confirmModalShow, setConfirmModalShow] = useState(false);
-  const handleClose = () => setShow(false);
-  const handleShow = () => setShow(true);
-  const handleConfirmModalChange = () => setConfirmModalShow(!confirmModalShow);
-  const [open, setOpen] = useState(false);
-
-  const handleSave = () => {
-    setShow(false);
-    saveFormData();
-  };
 
-  useEffect(() => {
-    if (createDesigns && processListData?.parentFormId && !formHistory.length) {
-      getFormHistory(processListData?.parentFormId).then((res) => {
-        dispatch(setFormHistories(res.data));
-      }).catch(() => {
-        setFormHistories([]);
-      });
-    }
-  }, [processListData]);
-
-  useEffect(() => {
-    if (processListData?.description) {
-      setFormDescription(processListData?.description);
-    }
-  }, [processListData?.description]);
-
-  useEffect(() => {
-    if (restoredFormId) {
-      setCurrentFormLoading(true);
-      fetchFormById(restoredFormId)
-        .then((res) => {
-          if (res.data) {
-            const { data } = res;
-            dispatch(setRestoreFormData(res.data));
-            dispatchFormAction({
-              type: "components",
-              value: _cloneDeep(data.components),
-            });
-            dispatchFormAction({ type: "type", value: data.type });
-            dispatchFormAction({ type: "display", value: data.display });
-          }
-        })
-        .catch((err) => {
-          toast.error(err.response.data);
-        })
-        .finally(() => {
-          setCurrentFormLoading(false);
-        });
-    }
-    return () => {
-      dispatch(setRestoreFormData({}));
-      dispatch(setRestoreFormId(null));
-    };
-  }, [restoredFormId]);
-
-  //remove tenatkey form path name
-  useEffect(() => {
-    if (form?.path && MULTITENANCY_ENABLED) {
-      const newPath = removeTenantKey(form.path, tenantKey);
-      if (newPath) {
-        dispatchFormAction({ type: "path", value: newPath });
-      }
-    }
-  }, [form?.path]);
-  // remove tenant key from form name
-  useEffect(() => {
-    if (form?.name && MULTITENANCY_ENABLED) {
-      const newName = removeTenantKey(form.name, tenantKey);
-      if (newName) {
-        dispatchFormAction({ type: "name", value: newName });
-      }
-    }
-  }, [form?.name]);
-
-  // setting the form data
-  useEffect(() => {
-    const newForm = formData;
-    if (
-      newForm &&
-      (form._id !== newForm._id || form.modified !== newForm.modified)
-    ) {
-      dispatchFormAction({ type: "replaceForm", value: newForm });
+  //for save form 
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const formAccess = useSelector((state) => state.user?.formAccess || []);
+  const submissionAccess = useSelector((state) => state.user?.submissionAccess || []);
+  const prviousData = useSelector((state) => state.process?.formPreviousData);
```

Comment:
Change variable name **prviousData** to **previousData** everywhere

Reply:
The issue came from the previous code.